### DISCUSSION TITLE
Run Avalonia App with Admin Permissions.

### DISCUSSION THREAD

**Discussion by ShrutiJaiswal1494** [AvaloniaUI/Avalonia#6607] [Q&A]
Hi,

My application needs Admin permissions to run. I need a code solution where the application can elevate to admin permissions on its own (any sample code will be excellent) so that the user does not have to give admin permissions every time. I do find some solutions for Windows but I am looking for something that will work on all the platforms. Any help is much appreciated. Thanks in advance.

**Comment by kekekeks** [comment]
On Linux you need to find if any of privilege elevation tools are installed (kdesudo, gksudo) and them use them to ask the user to run your executable as root.

You are not expected to run the app itself as root on macOS, you need to run a UI-less background app instead. That can be achieved in a similar way to Linux via applescript `osascript -e "do shell script \"/path/to/your/app\" with administrator privileges"`

**Reply by ShrutiJaiswal1494** [reply]
@kekekeks Would you please suggest anything for Windows as well?

**Reply by kekekeks** [reply]
For Windows the process is the same as for any other .NET Core app, you can follow any existing guide like this one: https://stackoverflow.com/a/61174964/2231814

Make sure to read about [application manifests on MSDN](https://docs.microsoft.com/en-us/windows/win32/sbscs/application-manifests) though.

**Reply by ShrutiJaiswal1494** [reply]
@kekekeks I did exactly this. This is working fine if I run the .exe file directly, it will show the UAC prompt every time but if I install the app and run it, it gives me an error like `The request is not supported`. I am not sure if this is something wrong with the Windows.

![image](https://user-images.githubusercontent.com/79670853/133844573-b190aa32-60e3-4f7c-9091-60c9049e14eb.png)

**Reply by maxkatz6** [reply]
Do you use MSIX app container?
In that case you need to add runFullTrust and allowElevation capabilities to the msix manifest (another manifest file).

See: 
https://stefanwick.com/2018/10/01/app-elevation-samples-part-1/
https://docs.microsoft.com/en-us/windows/uwp/packaging/app-capability-declarations

Keep in mind that Microsoft does not recommend it (apps with admin permissions) and it can be problematic with MS Store submission (from docs page):
> We don't recommend that you declare this capability in applications that you submit to the Microsoft Store. In most cases, the use of this capability won't be approved. It will only be approved for line-of-business apps deployed by enterprises to their private store via the Microsoft Store for Business.

**Reply by maxkatz6** [reply]
Also, it doesn't depend on Avalonia, so you can search for information without Avalonia keywords.
